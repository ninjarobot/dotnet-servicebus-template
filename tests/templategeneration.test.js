const Generator = require('@asyncapi/generator');
const path = require('path');
const { readFile } = require('fs').promises;

const {describe, test, expect} = require('@jest/globals');

describe("Template generation", () => {
    test('Generate deployment template from sb-end-to-end spec', async () => {
        const params = { location: 'eastus' };
        let outputDir =  path.resolve('tests', 'outputs')
        const generator = new Generator(path.normalize('./'), outputDir, { forceWrite:true, templateParams: params });
        await generator.generateFromFile(path.resolve('fsharp', 'AsyncApiServiceBusTests','sample-specs','sb-end-to-end.yml'));
        const deployment = await readFile(path.join(outputDir, 'deploy.json'), 'utf8');
        expect(deployment).toMatchSnapshot();
    });
});