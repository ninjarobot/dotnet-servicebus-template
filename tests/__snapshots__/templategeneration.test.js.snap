// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template generation Generate deployment template from sb-end-to-end spec 1`] = `
"{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "comment": "End to end Service Bus Example",
    "contentVersion": "1.0.0.0",
    "resources": [
        {
            "type": "Microsoft.ServiceBus/namespaces",
            "apiVersion": "2017-04-01",
            "location": "eastus",
            "name": "my-test-sbus",
            "sku": {
                "name": "Standard",
                "tier": "Standard"
            }
        },
        {
            "type": "Microsoft.ServiceBus/namespaces/topics",
            "apiVersion": "2017-04-01",
            "dependsOn": [
                "[resourceId('Microsoft.ServiceBus/namespaces', 'my-test-sbus')]"
            ],
            "name": "my-test-sbus/topic1"
        },
        {
            "type": "Microsoft.ServiceBus/namespaces/topics/subscriptions",
            "apiVersion": "2017-04-01",
            "dependsOn": [
                "[resourceId('Microsoft.ServiceBus/namespaces/topics', 'my-test-sbus', 'topic1')]"
            ],
            "name": "my-test-sbus/topic1/sub1"
        }
    ]
}"
`;
